docs/_spec/08-pattern-matching.md

```diff
@@ -307,6 +309,41 @@ All alternative patterns are type checked with the expected type of the pattern.
 They may not bind variables other than wildcards.
 The alternative pattern matches a value ´v´ if at least one its alternatives matches ´v´.
 
+### Given patterns
+
+```ebnf
+  GivenPattern   ::= TypePat
+```
+
+A _given pattern_ introduces an _anonymous given instance_ of the _type_ matched by `TypePat` to the scope of the clause that a given pattern occurs _within_.
+
+The rules for the _name_ synthesized for each anonymous instance is defined in the reference. If multiple names synthesized by the compiler are _not_ unique, this will result in an error.
```

Comment:
I have not put a link here, because the implicit section has not been updated to reflect `given`/`using` yet and I do not know whether I can link to https://docs.scala-lang.org/scala3/reference/contextual/relationship-implicits.html.